src/map_extras.cpp

```diff
@@ -49,41 +51,202 @@ void mx_null( map &, const tripoint & )
         for( int y = 0; y < SEEY * 2; y++ ) {
-            if( x >= cx - 4 && x <= cx + 4 && y >= cy - 4 && y <= cy + 4 ) {
-                if( !one_in( 5 ) ) {
+            if( m.veh_at( tripoint( x,  y, abs_sub.z ) ) &&
+                m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+            } else {
+                if( x >= cx - dice( 1, 5 ) && x <= cx + dice( 1, 5 ) && y >= cy - dice( 1, 5 ) &&
+                    y <= cy + dice( 1, 5 ) ) {
+                    if( one_in( 7 ) && m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                        m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                    }
+                }
+                if( x >= cx - dice( 1, 6 ) && x <= cx + dice( 1, 6 ) && y >= cy - dice( 1, 6 ) &&
+                    y <= cy + dice( 1, 6 ) ) {
+                    if( !one_in( 5 ) ) {
+                        m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    } else if( m.is_bashable( x, y ) ) {
+                        m.destroy( tripoint( x,  y, abs_sub.z ), true );
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    }
+
+                } else if( one_in( 4 + ( abs( x - cx ) + ( abs( y -
+                                         cy ) ) ) ) ) { // 1 in 10 chance of being wreckage anyway
                     m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
-                } else if( m.is_bashable( x, y ) ) {
-                    m.destroy( tripoint( x,  y, abs_sub.z ), true );
+                    if( !one_in( 3 ) ) {
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    }
                 }
-            } else if( one_in( 10 ) ) { // 1 in 10 chance of being wreckage anyway
-                m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
             }
         }
     }
 
-    m.spawn_item( rng( 5, 18 ), rng( 5, 18 ), "black_box" );
-    m.place_items( "helicopter", 90, cx - 4, cy - 4, cx + 4, cy + 4, true, 0 );
-    m.place_items( "helicopter", 20, 0, 0, SEEX * 2 - 1, SEEY * 2 - 1, true, 0 );
-    items_location extra_items = "helicopter";
-    switch( rng( 1, 4 ) ) {
+    int dir1 = rng( 0, 359 );
+    int crash_type = dice( 1, 9 );
+
+    std::enable_if<true, std::unique_ptr<vehicle>>::type veh;
+    switch( crash_type ) {
```

Comment:
You might consider creating a new helicopter wreckage [vehicle group](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/vehicle_groups.json), adding the helicopter wrecks to it, and then spawning a random one from the group. I'm not seeing anything in this that would require the hardcoded list, and that would make it possible for additional wreckage to be added in the future via json modification only.